.github/workflows/ci.yml

```diff
@@ -555,7 +555,7 @@ jobs:
       - run: python3 -m nox -s test
 
   test-free-threaded:
-    if: ${{ contains(github.event.pull_request.labels.*.name, 'CI-build-full') || github.event_name != 'pull_request' }}
+    if: github.ref != 'refs/heads/main'
```

Comment:
I think we do want to run this on main, because that's necessary to populate the GitHub Actions cache (don't ask me why they made it this way, it's their compute that we have to waste 😂).

```suggestion
```

Reply:
(It's for security reasons: You don't want PRs to pull caches from each _other_, otherwise one evil PR can poison the cache of a good PR, but it's fine to trust the cache of hte base branch.)

Reply:
Right, but I also believe that if the PR is merged it could be considered trustworthy and its cache contents could be imported into the base branch 😁